lib/ts/station/station.ex

```diff
@@ -16,7 +16,16 @@ defmodule Station do
     GenServer.start_link(Station, station_data)
   end
 
-  def init(station_data) do
+  def init(station_data = [station_struct, dependency]) do
```

Comment:
This code would be moved to station supervisor. Tests also become lot easier.

Reply:
Based on my reading of [Supervisor](https://hexdocs.pm/elixir/Supervisor.html) the children of the supervisor are defined in the following format.

Reply:
@SuhasHebbar The coding idiom mentioned by you is correct for specifying the child process. Here is the supervisor coding idiom taken from the supervisor hex doc.
```elixir
children = [
  # The Stack is a child started via Stack.start_link([:hello])
  %{
    id: Stack,
    start: {Stack, :start_link, [[:hello]]}
  }
]

# Now we start the supervisor with the children and a strategy
{:ok, pid} = Supervisor.start_link(children, strategy: :one_for_one)

#Now we can use the name / id and pid to register the child process. Such a registration happens inside the supervisor.
```